### ISSUE TITLE
Feature request: macro recorder

### ISSUE THREAD

**Issue by JohnTasler** [geddski/macros#12]
One thing I really like about most macro-enabled editors is the ability to "record" my action into a temporary macros, and then play it back multiple times.

This would be a great feature for you macros extension.

**Comment by geddski** [comment]
agreed! I'd be open to this as a community contribution, don't have time at the moment to add it myself.

**Comment by HaaLeo** [comment]
I think this will become possible as soon as the [`vscode.commands.onDidExecuteCommand`](https://code.visualstudio.com/updates/v1_37#_vscodecommandsondidexecutecommand) API is added to the stable release. 
Then as soon as the user activates recording (for instance via a command) the extension can listen to this event and create the corresponding macros command.
A drawback could be that this API will require the latest vscode version (probably ^1.38.0)

**Comment by HaaLeo** [comment]
Nevermind. Unfortunately the `vscode.commands.onDidExecuteCommand` API got rejected. See microsoft/vscode#78091
